src/libraries/Common/src/Interop/Windows/SspiCli/Interop.SSPI.cs

```diff
@@ -215,6 +215,92 @@ public enum Flags
             }
         }
 
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct SCH_CREDENTIALS
+        {
+            public const int CurrentVersion = 0x5;
+
+            public int dwVersion;
+            public int dwCredformat;
+            public int cCreds;
+
+            // This is pointer to arry of CERT_CONTEXT*
+            // We do not use it directly in .NET. Instead, we wrap returned OS pointer in safe handle.
```

Comment:
Is this comment needed?  I'm not even sure what it means; we do use it.

Reply:
the meaning was that we do not dereference or use CERT_CONTEXT in .NET. So it is pointer to a structure but that is hidden to PAL.

Reply:
I added that comment after your questions about IntPtr vs something strongly typed.

Reply:
So the "it" in we dot not use it" is referring to CERT_CONTEXT rather than to paCred?  That was my confusion.

But, even so, we are kind of using it, aren't we?  X509Certificate.Handle must be a CERT_CONTEXT*; we're storing that into paCred, effectively storing a pointer to an array of one CERT_CONTEXT.  So while we're not naming it as such, we are using it.

Reply:
yes. I assumed the "directly" would be a hint but obviously it is not clear.  I was also thinking about also adding CERT_CONTEXT so we can see details in debugger but that is somewhat complicated with more pointers inside.
Maybe Windbg could do cast to native but I failed so far with VS.